### PR TITLE
Boards: Update nucleo_f429zi to fit new default configuration guidelines

### PR DESCRIPTION
**Pull request by erwango**
This change aims at providing a reference for #7151 (update all zephyr boards according to default configuration guidelines).

- Set Arduino compatible instances for I2C, SPI, SERIAL and PWM
- Disable IPs by default
- Update board dts with arduino connectors
- Update board yaml file
- Update board documentation

Additionally, it contains 2 commits to enable treatment of device tree connector objects in dts parsing scripts.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
Doc changes LGTM.

**Comment by erwango** [comment]
recheck

**Comment by erwango** [comment]
recheck

**Comment by erwango** [comment]
recheck

**Comment by erwango** [comment]
@galak , am having issue with CI and I can't get log (no "download button"). Is there something I can do

File: scripts/dts/devicetree.py [hunk 1]
```diff
@@ -152,6 +160,9 @@ def parse_node(line, fd):
         'props': {},
         'name': build_node_name(name, addr)
     }
+    if alt_label:
+        node['alt_name'] = alt_label
```
Comment (therealprof) [inline scripts/dts/devicetree.py]: I find the mix of `alt_name` and `alt_label` mildly confusing, can't we use the same title for both?

File: scripts/dts/devicetree.py [hunk 2]
```diff
@@ -48,15 +48,23 @@ def parse_node_name(line):
         addr = None
 
     if ':' in line:
-        label, name = line.split(':')
+        if len(line.split(':')) == 3:
+            atl_label, label, name = line.split(':')
```
Comment (therealprof) [inline scripts/dts/devicetree.py]: Should be `alt_label` instead of `atl_label`.

Comment (erwango) [reply scripts/dts/devicetree.py]: Indeed, thanks

File: scripts/dts/devicetree.py (hunk 1 above)
Comment (erwango) [reply scripts/dts/devicetree.py]: For 'alt_label', I wanted to provide an equivalent to 'label' in this script.
Thought, for use in extract_dts_includes, I preferred the use of alt_name not to mix with 'label' which is an often given property. In extract_dts_includes alias is not an alternative to 'label'.

**Review by galak — APPROVED** [review]

**Pull request merged**